classes/watcher/login_watcher.php

```diff
@@ -0,0 +1,63 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_mfa\watcher;
+
+defined('MOODLE_INTERNAL') || die();
+
+use core\hook\after_require_login as after_require_login_hook;
+use core\hook\after_config as after_config_hook;
```

Comment:
It would be better to `use` these classes without aliasing them.

As per the Moodle coding standards here, this is discouraged and should only be used when resolving name conflicts: https://moodledev.io/general/development/policies/codingstyle#further-notes

> Do not alias class imports (use XXX as YYY;) unless it is absolutely required to resolve a conflict.

Reply:
Done, Aliasing is generally bad in all languages, frameworks I've worked with but after_config is a terrible classname (regardless of the context provided by it's namespace).

Reply:
> after_config is a terrible classname

Totally